ValveResourceFormat/ThirdParty/DDSImage.cs

```diff
@@ -229,47 +229,76 @@ private static void DecompressBlockDXT5(int x, int y, int width, byte[] blockSto
                         }
                     }
 
-                    byte colorCode = (byte)((code >> 2 * (4 * j + i)) & 0x03);
-
+                    byte positionCode = (byte)((code >> 2 * (4 * j + i)) & 0x03);
                     int finalR = 0, finalG = 0, finalB = 0;
 
-                    switch (colorCode)
+                    if (color0 > color1)
```

Comment:
The color portions of DXT5 and DXT1 is identical and could also be unified. I just copied the DXT1 implementation because is was working correctly.